core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -971,18 +971,30 @@ private[spark] class TaskSchedulerImpl(
   }
 
   override def stop(): Unit = {
-    speculationScheduler.shutdown()
+    Utils.tryLogNonFatalError {
+      speculationScheduler.shutdown()
+    }
     if (backend != null) {
-      backend.stop()
+      Utils.tryLogNonFatalError {
+        backend.stop()
+      }
     }
     if (taskResultGetter != null) {
-      taskResultGetter.stop()
+      Utils.tryLogNonFatalError {
+        taskResultGetter.stop()
+      }
     }
     if (barrierCoordinator != null) {
-      barrierCoordinator.stop()
+      Utils.tryLogNonFatalError {
+        barrierCoordinator.stop()
+      }
+    }
+    Utils.tryLogNonFatalError {
+      starvationTimer.cancel()
+    }
+    Utils.tryLogNonFatalError {
+      abortTimer.cancel()
```

Comment:
Does `Timer.cancel()` also require `Utils.tryLogNonFatalError`?

Reply:
Fair point yeah we can back that part out it shouldn't throw unless it's null and it's a val so